File: output/condition.py


def lua_truthy(value):
    return ((value is not None) and (value is not False))
num = 42
t = None
flag = True
if lua_truthy((num > 40)):
    num = 32
    print('true 0')
else:
    print('false 0')
if lua_truthy(t):
    print('true 1')
else:
    print('false 1')
if lua_truthy(flag):
    if lua_truthy(t):
        print('true 2')
    else:
        print('false 2')
if lua_truthy((flag and num)):
    print('true 3')
if lua_truthy((not flag)):
    print('false 4')
if lua_truthy(''):
    print('string true')

File: example/condition.lua
num = 42
t = nil
flag = true

if num > 40 then
    num = 32
    print("true 0")
else
    print("false 0")
end

if t then
    print("true 1")
else
    print("false 1")
end

if flag then
    if t then
        print("true 2")
    else
        print("false 2")
    end
end

if flag and num then
    print("true 3")
end

if not flag then
    print("false 4")
end

if "" then
    print("string true")
end
